icc-x-api/utils/data-owner-util.ts

```diff
@@ -0,0 +1,12 @@
+import { Delegation } from '../../icc-api/model/Delegation'
+import { HealthcareParty } from '../../icc-api/model/HealthcareParty'
+import { Patient } from '../../icc-api/model/Patient'
+import { Device } from '../../icc-api/model/Device'
+
+export function hasAccessTo(dataOwner: Patient | HealthcareParty | Device, delegations: { [key: string]: Array<Delegation> }): boolean {
+  return !!dataOwner.id && !!delegations[dataOwner.id] && !!delegations[dataOwner.id].length
+    ? true
+    : !!(dataOwner as HealthcareParty).parentId &&
```

Comment:
In fact, you should even do a recursive to get the dataOwner of the parent, then its parent again, until we have a parentId equal to null